events/oio_events_queue_beanstalkd.c

```diff
@@ -264,17 +318,62 @@ _q_run (struct oio_events_queue_s *self, gboolean (*running) (gboolean pending))
 	struct _queue_BEANSTALKD_s *q = (struct _queue_BEANSTALKD_s *)self;
 	gchar *saved = NULL;
 	int fd = -1;
-	int try_count = 0;
-	gint64 last_flush = 0;
+	guint attempts_connect = 0, attempts_check = 0, attempts_put = 0;
+	gint64 last_flush = 0, last_check = 0;
 
 	EXTRA_ASSERT (q != NULL && q->vtable == &vtable_BEANSTALKD);
 	EXTRA_ASSERT (running != NULL);
 
 	while ((*running)(0 < g_async_queue_length(q->queue))) {
-		gint64 now = oio_ext_monotonic_time();
+
+		const gint64 now = oio_ext_monotonic_time();
+
+		/* Maybe reconnect to the beanstalkd */
+		if (fd < 0) {
+			GError *err = NULL;
+			fd = sock_connect(q->endpoint, &err);
+			if (!err)
+				err = _poll_out (fd);
+			if (!err) {
+				err = _use_tube (fd, q->tube);
+				if (intercept_errors)
+					(*intercept_errors) (err);
```

Comment:
Here you may want to prefix the error to say the TCP connection is OK but the tube selection failed. This could help in case of a TCP port mismatch where the listening service is not a beanstalkd.